base_external_api_ftp/models/apicli_connection_ftp.py

```diff
@@ -18,27 +18,36 @@ class ApicliConnection(models.Model):
         selection_add=[("ftp", "FTP")], ondelete={"ftp": "set default"}
     )
 
-    def delete_ftp_file(self, ftp_session, subdirectory):
+    def create_delete_file_message(self, ftp_session, subdirectory):
```

Comment:
Current structure will have performance issues.
As a first step to address that, please split the create and delete logic in two separate methods.
Thanks